jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js

```diff
@@ -50,64 +58,104 @@ class DeploymentDirectDeploy extends React.Component {
     });
   }
 
-  render() {
-    let diagram = null;
-    let titleAndDescription = null;
+  /**
+    Generate the entity link.
+
+    @returns {Object} The link markup or null.
+  */
+  _generateLink() {
     const ddEntityId = this.props.ddData.id;
+    let url;
+    try {
+      url = window.jujulib.URL.fromLegacyString(ddEntityId);
+    } catch(_) {
+      url = window.jujulib.URL.fromString(ddEntityId);
+    }
+    url = this.props.generatePath({
+      store: url.path()
+    });
+    if (!url) {
```

Comment:
I just moved this code from elsewhere in the file. It does use the `generatePath` from state so doesn't look like it can be falsey. I'll remove the condition.